Trappist/src/Promact.Trappist.Repository/Questions/QuestionRepository.cs

```diff
@@ -109,5 +90,33 @@ public QuestionRepository(TrappistDbContext dbContext)
             });
             return languageNameList;
         }
+
+        public async Task UpdateSingleMultipleAnswerQuestionAsync(int questionId, QuestionAC questionAC, string userId)
+        {
+            var updatedQuestion = Mapper.Map<QuestionDetailAC, Question>(questionAC.Question);
+            if (await _dbContext.Question.Where(x => x.Id == questionId).AnyAsync())
+            {
+                //Update common Question details
+                updatedQuestion.Id = questionId;
+                updatedQuestion.UpdatedByUserId = userId;
+                _dbContext.Question.Update(updatedQuestion);
+                await _dbContext.SaveChangesAsync();
+
+                //Update single/multiple Question and option
+                var optionToRemove = await _dbContext.SingleMultipleAnswerQuestionOption.Where(x => x.SingleMultipleAnswerQuestionID == questionId).ToListAsync();
+                _dbContext.SingleMultipleAnswerQuestionOption.RemoveRange(optionToRemove);
+                await _dbContext.SaveChangesAsync();
+                foreach (SingleMultipleAnswerQuestionOption option in questionAC.SingleMultipleAnswerQuestion.SingleMultipleAnswerQuestionOption)
+                {
+                    option.SingleMultipleAnswerQuestionID = questionId;
+                    await _dbContext.SingleMultipleAnswerQuestionOption.AddAsync(option);
+                }
+                await _dbContext.SaveChangesAsync();
+            }
+            else
+            {
+                throw new KeyNotFoundException("No question found");
```

Comment:
it should be handled in a controller. And in case of Question not found it should return 404